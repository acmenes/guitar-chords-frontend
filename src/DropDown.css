.dropdown-div {
  width: 60%;
  align-items: center;
  margin: auto;
  padding: 10px;
}

.form-option {
  padding: 5px;
}

.load-chord {
  margin-top: 10px;
}
