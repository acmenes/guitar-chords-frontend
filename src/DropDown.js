import Chord from "@tombatossals/react-chords/lib/Chord";
import React, { useState } from "react";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form"

import "./DropDown.css"

///don't keep pick a chord here
const roots = ["Pick a Chord", "G", "G#", "Ab", "A", "A#", "Bb", "B", "B#", "C", "C#",
                "Db", "D", "D#", "Eb", "E", "E#", "Fb", "F", "F#"];

const rootsItems = [];

///don't keep pick a quality here
const qualities = ["Pick a quality", "major", "minor", "7th", "maj7", "min7", "augmented", "diminished"];

const qualitiesItems = [];

function DropDownMenu({ setChordName, setChord }) {
    const [formData, setFormData] = useState({
        roots: "",
        qualities: "",
    });
   
    const handleChangeRoot = evt => {
        const { roots, value } = evt.target;
        alert("changed")
        setFormData(formData => ({
            ...formData,
            roots: value,
        }));
    };

    const handleChangeQuality = evt => {
        const { qualities, value } = evt.target;
        alert("changed")
        setFormData(formData => ({
            ...formData,
            qualities: value,
        }));
    };

    for (const [index, value] of roots.entries()) {
        rootsItems.push(<option value={formData.index}>{value}</option>)
    }

    for (const [index, value] of qualities.entries()) {
        qualitiesItems.push(<option value={formData.index}>{value}</option>)
    }

    const handleSubmit = evt => {
        evt.preventDefault();
        console.debug(formData)
        setChordName(`${formData.roots} ${formData.qualities} Chord`);
        setChord({
            ///just a test for now to see how the chords will pop up
            /// this data needs to come from the DB/API
            frets: [0, 0, 0, 2, 3, 2],
            fingers: [0, 0, 0, 1, 3, 2],
            barres: [],
            capo: false
        })
    }
    return(
        <div className="dropdown-div">
       <Form>
           <Form.Label htmlFor="root-select" className="form-option">Select Your Root</Form.Label>
           <Form.Select onChange={handleChangeRoot}>
               {rootsItems}
           </Form.Select>
       </Form>
       <Form>
           <Form.Label htmlFor="quality-select" className="form-option">Select Chord Type</Form.Label>
           <Form.Select onChange={handleChangeQuality}>
               {qualitiesItems}
           </Form.Select>
           <Button onClick={handleSubmit} className="load-chord">Load Chord</Button>
       </Form>
       </div>

    )    
};

export default DropDownMenu;