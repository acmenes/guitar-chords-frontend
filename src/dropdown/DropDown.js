import Chord from "@tombatossals/react-chords/lib/Chord";
import React, { useState } from "react";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";

import GuitarApi from "../Api";

import Alert from "../common/Alert";

import "./DropDown.css";

///don't keep pick a chord here
const roots = [
  "--Pick a chord--",
  "G",
  "G#",
  "Ab",
  "A",
  "A#",
  "Bb",
  "B",
  "B#",
  "Cb",
  "C",
  "C#",
  "Db",
  "D",
  "D#",
  "Eb",
  "E",
  "E#",
  "Fb",
  "F",
  "F#",
];

const rootsItems = [];

///don't keep pick a quality here
const qualities = [
  "--Pick a quality--",
  "major",
  "minor",
  "7",
  "maj7",
  "min7",
  "augmented",
  "diminished",
];

const qualitiesItems = [];

function DropDownMenu({ setChordName, setChord }) {
  const [formData, setFormData] = useState({
    roots: "",
    qualities: "",
  });

  const handleChangeRoot = (evt) => {
    const { roots, value } = evt.target;
    setFormData((formData) => ({
      ...formData,
      roots: value,
    }));
    submit(value, formData.qualities);
  };

  const handleChangeQuality = (evt) => {
    const { qualities, value } = evt.target;
    setFormData((formData) => ({
      ...formData,
      qualities: value,
    }));
    submit(formData.roots, value);
  };

  /// for some reason, each one of these loops through twice
  for (const [index, value] of roots.entries()) {
    rootsItems.push(<option value={formData.index}>{value}</option>);
  }

  for (const [index, value] of qualities.entries()) {
    qualitiesItems.push(<option value={formData.index}>{value}</option>);
  }

  const submit = (roots, qualities) => {
    console.debug(formData);
    setChordName(`${roots}-${qualities}`);
    // let chord = await GuitarApi.getChord()

    async function getChordFromApi() {
      let chord = await GuitarApi.getChord(
        `${roots.toLowerCase()}-${qualities}`
      );
      if (!chord.chord[0]) return;
      let chordData = chord.chord[0];
      console.log(chordData);

      /// this can later be turned into a hook?
      let frets = chordData.frets.split(",");
      let fretsArray = [];
      for (let x = 0; x < frets.length; x++) {
        fretsArray.push(parseInt(frets[x]));
      }

      let fingers = chordData.fingers.split(",");
      let fingersArray = [];
      for (let x = 0; x < fingers.length; x++) {
        fingersArray.push(parseInt(fingers[x]));
      }

      let barresArray = [];
      if (chordData.barres === "None") {
        barresArray.push();
      } else {
        let barres = chordData.barres.split(" ");
        for (let x = 0; x < barres.length; x++) {
          barresArray.push(parseInt(barres[x]));
        }
      }

      setChord({
        frets: fretsArray,
        fingers: fingersArray,
        barres: barresArray,
        capo: false,
      });
    }

    getChordFromApi();

    /// should display an alert if the chord isn't found
  };
  return (
    <div className="dropdown-div">
      <Form>
        <Form.Label htmlFor="root-select" className="form-option">
          Select Your Root
        </Form.Label>
        <Form.Select onChange={handleChangeRoot}>{rootsItems}</Form.Select>
      </Form>
      <Form>
        <Form.Label htmlFor="quality-select" className="form-option">
          Select Chord Type
        </Form.Label>
        <Form.Select onChange={handleChangeQuality}>
          {qualitiesItems}
        </Form.Select>
      </Form>
    </div>
  );
}

export default DropDownMenu;
