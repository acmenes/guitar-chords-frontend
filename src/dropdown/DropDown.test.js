import React from "react";
import { render } from "@testing-library/react";
import DropDown from "./DropDown";

it("matches snapshot", function () {
  const { asFragment } = render(<DropDown />);
  expect(asFragment()).toMatchSnapshot();
});