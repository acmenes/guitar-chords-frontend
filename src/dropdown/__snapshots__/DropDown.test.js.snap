// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown-div"
  >
    <form
      class=""
    >
      <label
        class="form-option form-label"
        for="root-select"
      >
        Select Your Root
      </label>
      <select
        class="form-select"
      >
        <option>
          --Pick a chord--
        </option>
        <option>
          G
        </option>
        <option>
          G#
        </option>
        <option>
          Ab
        </option>
        <option>
          A
        </option>
        <option>
          A#
        </option>
        <option>
          Bb
        </option>
        <option>
          B
        </option>
        <option>
          B#
        </option>
        <option>
          Cb
        </option>
        <option>
          C
        </option>
        <option>
          C#
        </option>
        <option>
          Db
        </option>
        <option>
          D
        </option>
        <option>
          D#
        </option>
        <option>
          Eb
        </option>
        <option>
          E
        </option>
        <option>
          E#
        </option>
        <option>
          Fb
        </option>
        <option>
          F
        </option>
        <option>
          F#
        </option>
      </select>
    </form>
    <form
      class=""
    >
      <label
        class="form-option form-label"
        for="quality-select"
      >
        Select Chord Type
      </label>
      <select
        class="form-select"
      >
        <option>
          --Pick a quality--
        </option>
        <option>
          major
        </option>
        <option>
          minor
        </option>
        <option>
          7
        </option>
        <option>
          maj7
        </option>
        <option>
          min7
        </option>
        <option>
          augmented
        </option>
        <option>
          diminished
        </option>
      </select>
    </form>
  </div>
</DocumentFragment>
`;
